File: app/channels/hi_channel.rb
class HiChannel < ApplicationCable::Channel
  def subscribed
    stream_from "some_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def send_message(data)
    ActionCable.server.broadcast "some_channel", data
  end
end

File: app/javascript/channels/hi_channel.js
import consumer from "./consumer"


consumer.subscriptions.create("HiChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    console.log("received", data)
    let tr = document.createElement('tr')
    tr.textContent = data['data']
    document.querySelector('tr:last-child').appendChild(tr)
  },

  send_message(data){
    this.perform("send_message", {data: data})
  }
});

window.consumer = consumer;
